oeq-ts-rest-api/src/AxiosInstance.ts

```diff
@@ -22,16 +22,26 @@ const catchHandler = (error: AxiosError | Error): never => {
  * @param queryParams The query parameters to send with the GET request
  * @param transformer A function which returns a copy of the raw data from the GET with any required values transformed - this should NOT mutate the input data (transforms should start on a copy/clone of the input)
  */
-export const GET = <T>(
+export function GET<T>(path: string): Promise<unknown>;
+export function GET<T>(
   path: string,
-  validator?: (data: unknown) => boolean,
+  validator: (data: unknown) => data is T,
   queryParams?: object,
-  transformer?: (data: unknown) => T,
-): Promise<T> =>
-  axios
-    .get(path, {params: queryParams, paramsSerializer: params => stringify(params)}, )
-    .then((response: AxiosResponse) => {
-      const data: any = transformer? transformer(response.data) : response.data;
+  transformer?: (data: unknown) => T
+): Promise<T>;
+export function GET<T>(
+  path: string,
+  validator?: (data: unknown) => data is T,
+  queryParams?: object,
+  transformer?: (data: unknown) => T
+): Promise<T> | Promise<unknown> {
```

Comment:
This typing is a union of the two overloads, the overloads the precedence.

also note that validator is now required to be a type predicate.

Reply:
The implementation must ensure return type covers all overloads. Do we have to return `any` if we have a lot more overloads?

Reply:
I have not seen a case where `any` is needed.